dnsutil/zone.go

```diff
@@ -0,0 +1,209 @@
+package dnsutil
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"sort"
+	"strings"
+
+	"github.com/miekg/dns"
+)
+
+type zoneGenData struct {
+	Origin     string
+	DefaultTtl uint32
+	Records    []dns.RR
+}
+
+func (z *zoneGenData) Len() int      { return len(z.Records) }
+func (z *zoneGenData) Swap(i, j int) { z.Records[i], z.Records[j] = z.Records[j], z.Records[i] }
+func (z *zoneGenData) Less(i, j int) bool {
+	a, b := z.Records[i], z.Records[j]
```

Comment:
to properly sort names you will need to resort to the wireformat, see https://github.com/miekg/dns/blob/dfefa36553955889dbb9ef892d4bd116ec095ce2/dnssec.go#L587

If think that something that *efficently* (is so far possible with go dns) sorts a list of RRs is useful